Description: Disable JIT on x32, needs explicit porting to ILP32 amd64, FPU-less PowerPC.
Author: Thorsten Glaser <tg@mirbsd.org>, Roland Stigge <stigge@antcom.de>

Index: pcre3/sljit/sljitConfigInternal.h
===================================================================
--- pcre3.orig/sljit/sljitConfigInternal.h	2015-07-31 12:00:43.000000000 +0100
+++ pcre3/sljit/sljitConfigInternal.h	2015-12-22 13:30:50.000000000 +0000
@@ -117,7 +117,11 @@
 #if defined(__i386__) || defined(__i386)
 #define SLJIT_CONFIG_X86_32 1
 #elif defined(__x86_64__)
-#define SLJIT_CONFIG_X86_64 1
+# if defined(__ILP32__)
+#  define SLJIT_CONFIG_UNSUPPORTED 1
+# else
+#  define SLJIT_CONFIG_X86_64 1
+# endif
 #elif defined(__arm__) || defined(__ARM__)
 #ifdef __thumb2__
 #define SLJIT_CONFIG_ARM_THUMB2 1
@@ -131,7 +135,11 @@
 #elif defined(__ppc64__) || defined(__powerpc64__) || defined(_ARCH_PPC64) || (defined(_POWER) && defined(__64BIT__))
 #define SLJIT_CONFIG_PPC_64 1
 #elif defined(__ppc__) || defined(__powerpc__) || defined(_ARCH_PPC) || defined(_ARCH_PWR) || defined(_ARCH_PWR2) || defined(_POWER)
-#define SLJIT_CONFIG_PPC_32 1
+# ifndef __NO_FPRS__
+#  define SLJIT_CONFIG_PPC_32 1
+# else
+#  define SLJIT_CONFIG_UNSUPPORTED 1
+# endif
 #elif defined(__mips__) && !defined(_LP64)
 #define SLJIT_CONFIG_MIPS_32 1
 #elif defined(__mips64)
